src/toil/server/wes/toil_backend.py

```diff
@@ -0,0 +1,335 @@
+        """ Get information about the Workflow Execution Service."""
+
+        state_counts = {state: 0 for state in (
+            "QUEUED", "INITIALIZING", "RUNNING", "COMPLETE", "EXECUTOR_ERROR", "SYSTEM_ERROR", "CANCELING", "CANCELED"
+        )}
+        for _, state in self.get_runs():
+            state_counts[state] += 1
+
+        engine_parameters = []
+        for option in self.options:
+            if '=' not in option:  # flags like "--logDebug"
+                k, v = option, None
+            else:
+                k, v = option.split('=', 1)
+            engine_parameters.append((k, v))
+
+        return {
+            "version": baseVersion,
+            "workflow_type_versions": {
+                k: {
+                    "workflow_type_version": v
+                } for k, v in self.supported_versions.items()
+            },
+            "supported_wes_versions": ["1.0.0"],
+            "supported_filesystem_protocols": ["file", "http", "https"],
+            "workflow_engine_versions": {"toil": baseVersion},
+            "default_workflow_engine_parameters": [
+                {
+                    "name": key,
+                    "default_value": value
+                }
+                for key, value in engine_parameters
+            ],
+            "system_state_counts": state_counts,
+            "tags": {},
+        }
+
+    @handle_errors
+    def list_runs(self, page_size: Optional[int] = None, page_token: Optional[str] = None) -> Dict[str, Any]:
+        """ List the workflow runs."""
+        # TODO: implement pagination
+        return {
+            "workflows": [
+                {
+                    "run_id": run_id,
+                    "state": state
+                } for run_id, state in self.get_runs()
+            ],
+            "next_page_token": ""
+        }
+
+    @handle_errors
+    def run_workflow(self) -> Dict[str, str]:
+        """ Run a workflow."""
+        run_id = uuid.uuid4().hex
+        job = self._get_run(run_id, should_exists=False)
+
+        # set up necessary directories for the run
+        job.set_up_run()
+
+        # stage the uploaded files to the execution directory, so that we can run the workflow file directly
+        temp_dir = job.exec_dir
+        try:
+            _, request = self.collect_attachments(run_id, temp_dir=temp_dir)
+        except ValueError:
+            job.clean_up()
+            raise
+
+        wf_type = request["workflow_type"].lower().strip()
+        version = request["workflow_type_version"]
+
+        # validate workflow request
+        supported_versions = self.supported_versions.get(wf_type, None)
+        if not supported_versions:
+            job.clean_up()
+            raise RuntimeError(f"workflow_type '{wf_type}' is not supported.")
+        if version not in supported_versions:
+            job.clean_up()
+            raise RuntimeError("workflow_type '{}' requires 'workflow_type_version' to be one of '{}'.  Got '{}'"
```

Comment:
Same here, create custom Errors to replace Runtime Errors.  Something like `VersionNotImplemented`.